src/main/java/seedu/address/logic/commands/event/EEditCommand.java

```diff
@@ -0,0 +1,306 @@
+            setAddress(toCopy.address);
+            setZoomLink(toCopy.zoom);
+            setTags(toCopy.tags);
+            setTagsToDelete(toCopy.tagsToDelete);
+            setShouldDeleteAllTags(toCopy.shouldDeleteAllTags);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(name, start, end, description, address, zoom, tags, tagsToDelete)
+                    || shouldDeleteAllTags;
+        }
+
+        public void setName(Name name) {
+            this.name = name;
+        }
+
+        public Optional<Name> getName() {
+            return Optional.ofNullable(name);
+        }
+
+        public void setStartDateTime(StartDateTime start) {
+            this.start = start;
+        }
+
+        public Optional<StartDateTime> getStartDateTime() {
+            return Optional.ofNullable(start);
+        }
+
+        public void setEndDateTime(EndDateTime end) {
+            this.end = end;
+        }
+
+        public Optional<EndDateTime> getEndDateTime() {
+            return Optional.ofNullable(end);
+        }
+
+        public void setDescription(Description description) {
+            this.description = description;
+        }
+
+        public Optional<Description> getDescription() {
+            return Optional.ofNullable(description);
+        }
+
+        public void setAddress(Address address) {
+            this.address = address;
+        }
+
+        public Optional<Address> getAddress() {
+            return Optional.ofNullable(address);
+        }
+
+        public void setZoomLink(ZoomLink zoom) {
+            this.zoom = zoom;
+        }
+
+        public Optional<ZoomLink> getZoomLink() {
+            return Optional.ofNullable(zoom);
+        }
+
+        /**
+         * Sets {@code tags} to this object's {@code tags}.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        public Optional<Set<Tag>> getTags() {
+            return Optional.ofNullable(tags);
+        }
+        /**
+         * Returns an unmodifiable tag set to be removed, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tagsToDelete} is null.
+         */
+        public Optional<Set<Tag>> getTagsToDelete() {
+            return tagsToDelete != null ? Optional.of(Collections.unmodifiableSet((tagsToDelete))) : Optional.empty();
+        }
+
+        /**
+         * Sets {@code tagsToDelete} to this object's {@code tagsToDelete}.
+         * A defensive copy of {@code tagsToDelete} is used internally.
+         */
+        public void setTagsToDelete(Set<Tag> tagsToDelete) {
+            this.tagsToDelete = tagsToDelete != null ? new HashSet<>(tagsToDelete) : null;
```

Comment:
```suggestion
            this.tagsToDelete = (tagsToDelete != null && !tagsToDelete.isEmpty()) ? new HashSet<>(tagsToDelete) : null;
```

Do the same for `setTags` above. To standardize with `EditContactDescriptor`. We may need to implement a superclass to reduce such code duplication but for now, need to change.